src/fdc/DiskImageUtils.cc

```diff
@@ -189,34 +192,42 @@ static SetBootSectorResult setBootSector(
 	unsigned nbRootDirSectors = nbDirEntry / 16;
 	unsigned rootDirStart = nbReservedSectors + nbFats * nbSectorsPerFat;
 	unsigned firstDataSector = rootDirStart + nbRootDirSectors;
-	return {nbReservedSectors, firstDataSector, descriptor};
+	return {nbFats, nbSectorsPerFat, nbReservedSectors, rootDirStart, firstDataSector, descriptor};
 }
 
 void format(SectorAccessibleDisk& disk, bool dos1)
 {
 	// first create a boot sector for given partition size
 	size_t nbSectors = disk.getNbSectors();
 	SectorBuffer buf;
-	auto [firstFATSector, firstDataSector, descriptor] =
+	auto [fatCount, sectorsPerFat, fatStart, rootDirStart, dataStart, descriptor] =
```

Comment:
Similar here. It's fine for a few variables, but maybe not for 6.
I'd suggest you do `auto bootInfo = setBootSector(...);` and then use stuff like `bootInfo.fatCount`.

Hehe, you also swapped the first two here, so the error cancels out ;-)